20240808-custom-limit-email-notifications.md

```diff
@@ -28,22 +28,33 @@ Pro users are currently able to set customized project limits for storage and eg
 
 The purpose of "Project Limit Notification Emails" is to inform users automatically when they are getting close to, or exceeding their custom limits.
 
+In addition to notifications about custom limits, we want to notify users when they get close to or exceed their segment limits.
+
 ### Goals
 
 1. Give users advance notice that they are getting close to their custom limits (80% used)
 2. Inform users when they have exceeded their custom limits (100% used)
 
 ### Approach / Design
 
-Things we want to avoid:
+**Things we want to avoid**
 * unintentionally sending multiple emails per "threshold event" (e.g. storage usage on project abc has exceeded 80% of the custom limit)
 * failing to send an email for a "threshold event" (e.g. satellite restart at the same time as threshold event, resulting in dropped communications)
-
-These constraints require that we make some DB updates.
+* too long of a delay between occurrence of event and notification (e.g. >30m is probably too long, and 30m may even be too long)
+* significantly impacting metainfo performance
 
 #### Config Updates
 
-A single feature flag should be added on the satellite to enable "limit threshold email notifications". It should default to be turned off, until it is turned on in production.
+A feature flag should be added on the satellite to enable "limit threshold email notifications". It should default to be turned off, until it is turned on in production. This flag enables/disables:
```

Comment:
Should this be limited to specific projects? e.g. turning it on and it breaking on all projects could send out a lot of emails.

Reply:
Thinking about it more, I agree that only being able to enable or disable it globally could send out too many emails.
We could make it an opt-in feature on the project level. E.g. a box you check when setting custom limits or something you can find on the project settings page. Do you think that would address the concern?

Reply:
Yes, that would be fine. Although, I suspect most projects want to enable it rather than keep it disabled.